src/App.tsx

```diff
@@ -15,48 +16,99 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum SortType {
+  name = 'name',
+  length = 'length',
+  empty = ' ',
+}
+
+function getSortedGoods(
+  goods: string[],
+  sortBy: SortType,
+  reverse: boolean,
+) : string[] {
+  const preparedGoods = [...goods];
+
+  switch (sortBy) {
+    case SortType.name:
+      preparedGoods.sort((good1, good2) => good1.localeCompare(good2));
+      break;
+
+    case SortType.length:
+      preparedGoods.sort((good1, good2) => good1.length - good2.length);
+      break;
+
+    default:
+      break;
+  }
+
+  return reverse ? preparedGoods.reverse() : preparedGoods;
+}
+
+export const App = () => {
+  const [sortBy, setSortBy] = useState(SortType.empty);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const visibleGoods = getSortedGoods(
+    goodsFromServer, sortBy, isReversed,
+  );
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={cn('button', 'is-info', {
+            'is-light': sortBy !== SortType.name,
+          })}
+          onClick={() => {
+            setSortBy(SortType.name);
+          }}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={cn('button', 'is-success', {
+            'is-light': sortBy !== SortType.length,
+          })}
+          onClick={() => {
+            setSortBy(SortType.length);
+          }}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={cn('button', 'is-warning', {
+            'is-light': !isReversed,
+          })}
+          onClick={() => setIsReversed(prev => !prev)}
         >
           Reverse
         </button>
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {(sortBy !== SortType.empty || isReversed)
+          && (
+            <button
+              type="button"
+              className="button is-danger is-light"
+              onClick={() => {
+                setSortBy(SortType.empty);
+                setIsReversed(false);
+              }}
+            >
```

Comment:
NOTE: using anonymous function is bad pratice - you should create separate handler and pass to onClick prop reference to it